src/btree/col_modify.c

```diff
@@ -147,7 +147,8 @@ __wt_col_modify(WT_CURSOR_BTREE *cbt, uint64_t recno, const WT_ITEM *value, WT_U
         upd->next = old_upd;
 
         /* Serialize the update. */
-        WT_ERR(__wt_update_serial(session, page, &cbt->ins->upd, &upd, upd_size, false));
+        WT_ERR(__wt_update_serial(session, cbt, page, &cbt->ins->upd, &upd, upd_size, false,
+          btree->type == BTREE_COL_VAR && cbt->ins == NULL));
```

Comment:
Style-wise that's become too many parameters to a function call. It also doesn't feel right layering wise to have __wt_update_serial be told about where to look for things to check validity of updates.

Reply:
Removed check_onpage_value flag from __wt_update_serial